subt/tools/submit.py

```diff
@@ -0,0 +1,155 @@
+import argparse
+import json
+import re
+import requests
+import subprocess
+import os
+import pathlib
+
+import toml
+
+WORLDS=dict(
+    up1 = "Urban Practice 1",
+    up2 = "Urban Practice 2",
+    up3 = "Urban Practice 3",
+    cq  = "Cave Qualification",
+    cs1 = "Cave Simple 1",
+    cs2 = "Cave Simple 2",
+    cs3 = "Cave Simple 3",
+    cp1 = "Cave Practice 1",
+    cp2 = "Cave Practice 2",
+    cp3 = "Cave Practice 3",
+)
+
+ROBOTS=dict(
+    teambase = "TEAMBASE",
+    drone = "SSCI_X4_SENSOR_CONFIG_2",
+    freya = "ROBOTIKA_FREYJA_SENSOR_CONFIG_2",
+)
+
+
+def _cmd(*args):
+    return subprocess.run(args, stdout=subprocess.PIPE).stdout
+
+
+def validate_world(world):
+    try:
+        return WORLDS[world]
+    except KeyError:
+        sys.exit(f"'{world}' not valid world identification")
+
+
+def validate_robots(robots, timeout):
+    valid = {}
+    for name, kind in robots.items():
+        try:
+            valid[name] = ROBOTS[kind]
+        except KeyError:
+            sys.exit(f"'{kind}' not valid robot kind identification")
+    valid[f"T{timeout}"] = "TEAMBASE"
+    return valid
+
+
+def validate_image(image):
+    image_bytes = bytes(image, 'ascii')
+    images = _cmd('aws', 'ecr', 'list-images', '--repository-name', 'subt/robotika')
+    for img in images.split(b'\n'):
+        parts = img.split(b'\t')
```

Comment:
you have probably different output format, because if I add `print(parts)` I get:
```
[b'            "imageDigest": "sha256:86040c1fac4f5c3bbda3e3b766dd0504467e475a20e8c86b1e5905511229c5c1",']
[b'            "imageTag": "ver68fly"']
...
```

Reply:
my JSON output looks like in
https://docs.aws.amazon.com/cli/latest/reference/ecr/list-images.html
... so it would be maybe easier to parse JSON directly??

Reply:
I would love to do that but my output is not json and I am not sure why. Do yo get json output? What version of aws cli do you have? This is the output I am getting:

<pre>$aws ecr list-images --repository-name subt/robotika
IMAGEIDS	sha256:b728df56f42b2df2d4a4a87c901da77e6963faca38414d3e4802a29aa185a7ad	ver78mdbread
IMAGEIDS	sha256:c4f91dc814cb462833c8eb667d853fd228cf4d293891d2c8fd5eedaf6cac8d3f	ver81rc1
IMAGEIDS	sha256:6e972819a0260456601619a40971cc35f60d19983e891d205d66bdb925bf6dab	ver20
IMAGEIDS	sha256:8df3f91f76b58a3bde95eb534e36c2eb640fe253bcbe65b3f8b2ff637a754cce	zw004neighbors2
IMAGEIDS	sha256:fb0fe276a49e7d12d9bcde1d341e12cb4529fd7296e6a9451e9310a0f82581b3	seed2
IMAGEIDS	sha256:e2f8a9b83a599e5acabe7aa2754a92a7711e1bb7e1d62daa26cb5dd0ff034da8	ver1
....
</pre>

<pre>$ aws --version
aws-cli/1.18.105 Python/3.6.9 Linux/4.15.0-118-generic botocore/1.17.28
</pre>

Reply:
`--output json` is the magic argument

Reply:
at the end of the day, this might be even better option: https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/ecr.html#ECR.Client.list_images (in the future)

Reply:
Switched to json in 7611ad8a83b831199596b1c26541512c5517992a

Reply:
I must have changed the default some time ago:

<pre>$ cat .aws/config 
[default]
region = us-east-1
output = text
</pre>
https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-profiles.html
https://docs.aws.amazon.com/cli/latest/userguide/cli-usage-output.html

Reply:
No, I might just have selected something different from you during [aws configure](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-quickstart.html). Good to know & have it fixed. Thanks.